From 6680aafdfb7a2ec197a7040e504da72623967d4d Mon Sep 17 00:00:00 2001
From: Alexi Demers <alexidemers@gmail.com>
Date: Tue, 30 Jan 2024 20:29:04 -0500
Subject: [PATCH] rockchip-common: configure boot commands and apparmor=0

apparmor is not enabled, and it must be explicited to avoid k3s from
trying to use it.

This commit also sets the correct boot command to boot.
---
 include/configs/rockchip-common.h | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/include/configs/rockchip-common.h b/include/configs/rockchip-common.h
index e5b8368..5c45f00 100644
--- a/include/configs/rockchip-common.h
+++ b/include/configs/rockchip-common.h
@@ -135,15 +135,13 @@
 		"setenv devtype ramdisk; setenv devnum 0;" \
 	"fi; \0"
 
-#if defined(CONFIG_AVB_VBMETA_PUBLIC_KEY_VALIDATE)
-#define RKIMG_BOOTCOMMAND			\
-	"boot_android ${devtype} ${devnum};"
-#else
-#define RKIMG_BOOTCOMMAND			\
-	"boot_android ${devtype} ${devnum};"	\
-	"bootrkp;"				\
-	"run distro_bootcmd;"
-#endif
+#define RKIMG_BOOTCOMMAND		\
+	"setenv bootargs \"${bootargs} apparmor=0 root=/dev/mmcblk0p4\"; \
+	ext2load mmc 1:3 0x00280000 Image-initramfs-orangepi-r1plus-lts.bin; \
+	ext2load mmc 1:3 0x00280000 Image-initramfs-orangepi-r1plus.bin; \
+	ext2load mmc 1:3 0x08300000 rk3328-orangepi-r1-plus-lts.dtb; \
+	ext2load mmc 1:3 0x08300000 rk3328-orangepi-r1-plus.dtb; \
+	booti 0x00280000 - 0x08300000;"
 
 #endif /* CONFIG_SPL_BUILD */
 
-- 
2.34.1

